### PR TITLE
Fixed TexturedParallelepiped proto

### LINKED ISSUE #2361: Textured Parallelipiped checkboxes don't match cube mapping
Create a textured parallelipiped, select cube mapping, select the Webots-suggested cube_mapping.jpg as the texture to map.  The parallelipiped appears reasonably with the expected faces adjoining.  Now change the setting for _right_ to FALSE.

**Expected behavior:**  The blue side labelled RIGHT should be the side that disappears.  
**What actually happens:** the side labeled FRONT disappears instead.  (Similar problems affect several of the other checkboxes. 
 E.g. the side I wanted to remove was labelled BACK, but to remove it I actually had to set _left_ to FALSE.)

I'm not sure what to suggest to fix this mismatch, as any change you make may break backwards compatibility for some existing uses.  If you do change it, I guess I'd suggest changing which faces the checkboxes remove, rather than changing how the texture maps to the vertices, as it'd be annoying to have to use different cubemaps for these from what you use for textured boxes, and if the change does break backwards compatibility for someone, it's generally much easier to change a couple of boolean checkboxes than it is to make new texture maps.  Or maybe you should just clearly document the mismatch and live with it?

image

### PR DESCRIPTION
**Pull request by omichel**
Fixes #2361.

### CODE REVIEW

**Comment by DavidMansolino** [comment]
:eyes:

**Review by DavidMansolino — CHANGES_REQUESTED** [review]
It works now as expected, but as the user said, this clearly breaks the compatibility.
Before:
tmp2-2

Now: 
tmp2

This should therefore at least be written in bold in the changelog (I would even have targeted develop).

File: docs/reference/changelog-r2020.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Released on XXX YYY, 2020.
   - Enhancements
     - Added support for Python 3.9 ([#2318](https://github.com/cyberbotics/webots/pull/2079)).
   - Bug fixes
+    - **Fixed cube, compact and flat texture mappings of [TexturedParallelepiped](https://cyberbotics.com/doc/guide/object-geometries#texturedparallelepiped) proto** ([#2364](https://github.com/cyberbotics/webots/pull/2364)).
```
Comment (DavidMansolino) [inline docs/reference/changelog-r2020.md:9]:
```suggestion
    - **Fixed cube, compact and flat texture mappings of [TexturedParallelepiped](object-geometries.md#texturedparallelepiped) proto** ([#2364](https://github.com/cyberbotics/webots/pull/2364)).
```

**Comment by omichel** [comment]
I updated the change log with a bold sentence. However, I believe it is clearly a bug fix and as many bug fixes, it indeed may break the backwards compatibility. However, this primitive doesn't seem to be very much used: none of our PROTO or world file use it except for the `textured_proto_shape.wbt` sample world. So, the risk of breaking user simulations is low and in case it would happen, the fix is very simple. So, I would leave this bug fix on the master branch.

File: docs/reference/changelog-r2020.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ Released on XXX YYY, 2020.
   - Enhancements
     - Added support for Python 3.9 ([#2318](https://github.com/cyberbotics/webots/pull/2079)).
   - Bug fixes
+    - **Fixed cube, compact and flat texture mappings of [TexturedParallelepiped](object-geometries.md#texturedparallelepiped) proto** ([#2364](https://github.com/cyberbotics/webots/pull/2364)).
```
Comment (DavidMansolino) [inline docs/reference/changelog-r2020.md:9]:
```suggestion
    - **Fixed cube, compact and flat texture mappings of [TexturedParallelepiped](../guide/object-geometries.md#texturedparallelepiped) proto** ([#2364](https://github.com/cyberbotics/webots/pull/2364)).
```
Sorry for this.

**Comment by DavidMansolino** [comment]
> However, this primitive doesn't seem to be very much used: none of our PROTO or world file use it except for the textured_proto_shape.wbt

I am not sure about this, it is used directly in 3 PROTO files (and even more indirectly), and indirectly in a lot of worlds.

But that's fine, most of the case the texture is face independent.

**Review by DavidMansolino — APPROVED** [review]
Thank you.

**Comment by omichel** [comment]
Sorry, the meant the problematic texture mappings (cube, compact and flat) seems to be never used in our PROTO files.

**Pull request merged**